sdk/nodejs/automation/stack.ts

```diff
@@ -123,17 +123,17 @@ export class Stack {
         });
         await eventLogTail.start();
         const lineSplitter = readline.createInterface({ input: eventLogTail });
+        let partialLine = "";
         lineSplitter.on("line", (line) => {
             let event: EngineEvent;
             try {
-                event = JSON.parse(line);
+                line = partialLine + line;
+                partialLine = "";
+		event = JSON.parse(line);
                 callback(event);
             } catch (e) {
-                log.warn(`Failed to parse engine event
-If you're seeing this warning, please comment on https://github.com/pulumi/pulumi/issues/6768 with the event and any
-details about your environment.
-
-Event: ${line}\n${e.toString()}`);
+                partialLine += line;
+                return;
```

Comment:
Unfortunately `nodejs` doesn't seem to include the newline character in the on line events, so we'll have to go off catching exceptions here.